ui/packages/audio/src/Audio.svelte

```diff
@@ -73,38 +96,72 @@
 			}
 		}
 
-		if (stream === null) {
-			return;
-		}
+		if (stream == null) return;
 
-		recorder = new MediaRecorder(stream);
+		if (streaming) {
+			const [{ MediaRecorder, register }, { connect }] = await Promise.all(
+				module_promises
+			);
```

Comment:
We started the module fetching earlier on to avoid unnecessary delays but when someone presses record we need to make sure the modules have finished importing. We can do this by awaiting the promises we created earlier. If they are still loading them will will wait for them, then assign the exports we care about to these variables (we are deconstructing the arrays + objects).

If the promise has already resolved (imports loaded) then it will simply unpack the value and assign the parts we care about these values immediately.